src/middlewared/middlewared/plugins/jail.py

```diff
@@ -486,67 +526,60 @@ def _import(self, job, jail):
-
-            return pkgs
+            pkg_dict = self.middleware.call_sync('cache.get', 'iocage_rpkgdict')
+            r_plugins = self.middleware.call_sync('cache.get',
+                                                  'iocage_rplugins')
         except ClientException as e:
             # The jail plugin runs in another process, it's seen as a client
             if e.trace and e.trace['class'] == 'KeyError':
-                pass  # It's either new or past cache date
-            else:
-                raise(e)
-
-        r_pkgs = requests.get('http://pkg.cdn.trueos.org/iocage/All')
-        r_pkgs.raise_for_status()
-
-        r_plugins = requests.get(
-            'https://raw.githubusercontent.com/freenas/'
-            'iocage-ix-plugins/master/INDEX'
-        )
-        r_plugins.raise_for_status()
-        r_plugins = r_plugins.json()
-
-        pkgs = {
-            'bruserver': ('N/A', '1'),
-            'sickrage': ('Git branch - master', '1')
-        }
-        pkg_dict = {}
-
-        for i in r_pkgs.iter_lines():
-            i = i.decode().split('"')
+                r_pkgs = requests.get('http://pkg.cdn.trueos.org/iocage/All')
+                r_pkgs.raise_for_status()
+                pkg_dict = {}
+                for i in r_pkgs.iter_lines():
+                    i = i.decode().split('"')
+
+                    try:
+                        pkg, version = i[1].rsplit('-', 1)
+                        pkg_dict[pkg] = version
+                    except (ValueError, IndexError):
+                        continue  # It's not a pkg
+                self.middleware.call_sync(
+                    'cache.put', 'iocage_rpkgdict', pkg_dict,
+                    86400
+                )
 
-            try:
-                pkg, version = i[1].rsplit('-', 1)
-                pkg_dict[pkg] = version
-            except (ValueError, IndexError):
-                continue  # It's not a pkg
+                r_plugins = requests.get(
+                    'https://raw.githubusercontent.com/freenas/'
+                    'iocage-ix-plugins/master/INDEX'
+                )
+                r_plugins.raise_for_status()
 
-        for plugin, p_dict in r_plugins.items():
-            try:
-                primary_pkg = p_dict['primary_pkg'].split('/', 1)[-1]
-            except KeyError:
-                continue  # Plugin doesn't have a pkg, like bruserver
-
-            if primary_pkg in pkg_dict:
-                version = pkg_dict[primary_pkg]
-                pkgs[plugin] = (
-                    version.rsplit('%2', 1)[0].replace('.txz', ''),
-                    '1'
+                r_plugins = r_plugins.json()
+                self.middleware.call_sync(
+                    'cache.put', 'iocage_rplugins', r_plugins,
+                    86400
                 )
             else:
-                if plugin not in pkgs:
-                    pkgs[plugin] = ('N/A', 'N/A')
+                raise(e)
+
+        if pkg == 'bru-server':
```

Comment:
What is this?

Reply:
We have to hardcore two plugins for lack of a real pkg. This is a temporary bit of glue I plan to remove ASAP